src/pages/api/admin/updateTopicInfo.ts

```diff
@@ -0,0 +1,28 @@
+import { ClassModel } from '@server/db/class'
+import { apiController } from '@utils/api'
+import { Types } from 'mongoose'
+
+export interface UpdateTopicInfoParams {
+  cid: string
+  topics: string[]
+}
+
+export interface UpdateTopicInfoResults {
+  topics: string[]
+  success: boolean
+}
+
+export default apiController<UpdateTopicInfoParams, UpdateTopicInfoResults>(async ({ cid, topics }) => {
+  const CourseClass = await ClassModel.findById(new Types.ObjectId(cid))
+  if (CourseClass) {
+    const updateCourseTopics = await CourseClass.updateOne({ $set: { topics: topics } })
+    if (updateCourseTopics) {
+      const CourseClass = await ClassModel.findById(new Types.ObjectId(cid))
+      return {
+        topics: CourseClass.topics,
+        success: true,
```

Comment:
cuz success is true no matter what, it seems meaningless. If you want to give success status, consider using `throw new Error()`

Reply:
> cuz success is true no matter what, it seems meaningless. If you want to give success status, consider using `throw new Error()`

That does make sense I have updated it here https://github.com/kixlab/kuiz/pull/46/commits/aa327e685ddb33ab773f0a3120a65a78808094da